### ISSUE TITLE
Highstock time series does not draw end of the line for some charts

### ISSUE THREAD

**Issue by bradbesserman** [highcharts/highcharts#7831]
#### Issue
Highstock time series chart will occasionally not draw the end of the series.  This is fixed by resizing the chart.  See the GIF below for more detail.

#### Live demo with steps to [reproduce]
![highstock_issue](https://user-images.githubusercontent.com/29936283/36013476-6e0c7b5e-0d32-11e8-9c27-ee6e8b440fa2.gif)

This issue is difficult to reproduce.  It only occurs very occasionally, and only happens after I update the series.  It happens specifically when changing the yAxis labels from a label with a longer name to a label with a shorter name, and then updating the chart.  

#### Product version
Highcharts 6.0.3
This error is occurring using highcharts-angular.

#### Affected browser(s)
Chrome 63.0.3239.132

**Comment by KacperMadej** [comment]
Hi @robinsoncrus0e 

Thank you for reporting about the problem.
Please post a live demo with steps required to recreate the issue.

If you are not able to recreate the problem without the highcharts-angular you can post an issue directly on the wrapper's repository.

**Comment by pawelfus** [comment]
It can be connected to a few options:

- `series.dataGrouping` - could you check if disabling helps?
- `series.gapSize` - again, I would try to disable this option
- `xAxis.ordinal` -  it's also enabled by deafult

Simple demo works fine: http://jsfiddle.net/11sk0xcu/

Could you also check if chart's width should be fixed? Explanation how do you update the chart would be great too.

**Comment by robinsoncrus0e** [comment]
thanks for the response.  the suggested fixes do not change anything.  i've noticed that this issue does not happen when using the built in range selector.  in my application i have the range selector disabled and programatically set xmin and xmax.

**Comment by pawelfus** [comment]
@robinsoncrus0e thank you for the details. Maybe your extremes are wrong? Please, recreate issue in jsFiddle and we will check what happens there.

**Comment by pawelfus** [comment]
Hi @robinsoncrus0e 

Did you get a chance to recreate issue in jsFiddle?

**Comment by pawelfus** [comment]
Closing due to inactivity, please let us know when you will be able to recreate the issue - we will open this ticket.

**Comment by robinsoncrus0e** [comment]
@KacperMadej @pawelfus sorry it's taken me a while to get around to reproducing this bug.  I was able to reproduce this issue.  Go to the following website:

http://highcharts-bug.s3-website.us-east-2.amazonaws.com

If you click the "Toggle Data" button twice, the error appears on the rightmost part of the graph, near the labels.   

I reproduced this bug in Angular 5, and haven't tested if it also occurs in plain javascript.  I posted the bug here because it seems like highcharts-angular is just a basic wrapper for highcharts, and this seems to be a problem with the chart itself.  If you'd like me to post the bug on highcharts-angular instead, I can do that.

The bug occurs on Chrome, Firefox, and Edge.  The Angular app is using Highcharts 6.1.0 and highcharts-angular 1.0.2.

The chart options are as follows:

`chartOptions = {
    yAxis: [
      {
        labels: {
          align: "left"
        }
      }
    ],
    xAxis: [
      {
        min:  Date.now() - 60 * 60 * 24 * 1000,
      }
    ],
    series: [
      {
        data: [] // I fetch the data and add it later
      }
    ]
  };`

The button toggles between two different datasets by setting chartOptions.series[0].data to the new dataset.  I posted the code to Github - all of the relevant stuff can be found in app.component.ts and app.component.html.  This includes the datasets that the chart is toggling between.  Link to repo - https://github.com/robinsoncrus0e/highcharts-bug

#### Product version

**Comment by KacperMadej** [comment]
@robinsoncrus0e 
Thank you very much for the demo.

It took me some time to recreate without the Angular wrapper, but here it is: https://jsfiddle.net/BlackLabel/ysfxjhuy/1/

It looks like it's caused by updating chart while initial animation hasn't ended, because it's resolved by setting `animation` to `false` for the series - https://jsfiddle.net/BlackLabel/ysfxjhuy/2/

I have cloned your repo with the demo and applied the same changes - it works and the problem is resolved.

Another way for you to fix the problem in the Angular app and keep the animation is to set in the initial config in series options `animation` to `false` and when setting new data first time (in `ngOnInit` function) set also `animation` to `true` like:

```ts
    this.chartOptions.series[0].data = this.usdLine;
    this.chartOptions.series[0].animation = true;
    this.toggleData();
```

In summary, it's not recommended to set data just after chart was created - you could as well create a chart with the data set. However, for some situations when it's needed you could toggle animation as explained above.

Should this issue be reopened @robinsoncrus0e ?
